oeps/oep-0005.rst

```diff
@@ -0,0 +1,187 @@
+=======================================
+OEP-5: Pre-built Developer Environments
+=======================================
+
++---------------+-------------------------------------------+
+| OEP           | :doc:`OEP-5 </oeps/oep-0005>`             |
++---------------+-------------------------------------------+
+| Title         | Pre-built Developer Environments          |
++---------------+-------------------------------------------+
+| Last Modified | 2016-07-20                                |
++---------------+-------------------------------------------+
+| Author        | Calen Pennington <cale@edx.org>           |
++---------------+-------------------------------------------+
+| Arbiter       | Jesse Zoldak <jzoldak@edx.org>            |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Best Practice                             |
++---------------+-------------------------------------------+
+| Created       | 2016-06-28                                |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+Open edX is a large ecosystem of smaller services. In order to facilitate
+newcomers working in that system, each service will provide a prebuilt environment
+that is packaged, published, and ready to be used for development.
+
+Motivation
+==========
+
+Open edX and the edx-platform repository have gone through several iterations
+of developer setup instructions.
+
+The first set of instructions detailed how to install edx-platform and all of
+its dependencies on your local workstation. As the number of developers and
+developer environments increased, so did the workload of maintaining instructions
+to work around all of the idiosyncrasies in those environments. These instructions
+weren't useful for production installations, which were standardized on a
+particular release of the Ubuntu operating system.
+
+The second, current, method of development is referred to as Devstack. This is
```

Comment:
Sometimes its beneficial to have the high-level name not be directly
tied to the specifc implementation. In this case, the underlying
implemenation is kind of the point (docker vs vagrant), so I'm not
opposed to a name that references the supporting tech.

The first thing that pops to mind is `dockstack`.
`dock` implies a connection to docker.
`stack` implies that it's a collection of services.

It rolls off the tongue in a similar manner as `devstack` (two
syllables, same leading consonant on the prefix, etc.),
though maybe it's too similar.

That's just off-the-cuff; I'll put some more thought into it now too.